expression/integration_test.go

```diff
@@ -1753,3 +1753,41 @@ func (s *testIntegrationSuite) TestOtherBuiltin(c *C) {
 	result = tk.MustQuery("select (0,1) in ((0,1), (0,2)), (0,1) in ((0,0), (0,2))")
 	result.Check(testkit.Rows("1 0"))
 }
+
+func (s *testIntegrationSuite) TestDateBuiltin(c *C) {
+	defer func() {
+		s.cleanEnv(c)
+		testleak.AfterTest(c)()
+	}()
+
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("USE test;")
+	tk.MustExec("DROP TABLE IF EXISTS t;")
+	tk.MustExec("create table t (d date);")
+	tk.MustExec("insert into t values ('1997-01-02')")
+	tk.MustExec("insert into t values ('1998-01-02')")
+	r := tk.MustQuery("select * from t where d < date '1998-01-01';")
+	r.Check(testkit.Rows("1997-01-02"))
+
+	r = tk.MustQuery("select date'20171212'")
+	r.Check(testkit.Rows("2017-12-12"))
+
+	r = tk.MustQuery("select date'2017/12/12'")
+	r.Check(testkit.Rows("2017-12-12"))
+
+	r = tk.MustQuery("select date'2017/12-12'")
+	r.Check(testkit.Rows("2017-12-12"))
+
+	rs, _ := tk.Exec("select date'2017-99-99';")
+	_, err := tidb.GetRows(rs)
+	c.Assert(err, NotNil)
```

Comment:
1. if err is not nil, please add test to check error message
2. add test for zero date, like: "0000-00-00" or "0000-00-00 00:00:00"